File: manara/checkSubarraySum.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var checkSubarraySum = function (nums, k) {
  const remainders = { 0: -1 };
  let total = 0;

  for (const [i, num] of Object.entries(nums)) {
    total += num;
    const remainder = total % k;

    if (!remainders.hasOwnProperty(remainder)) {
      remainders[remainder] = i;
    } else if (i - remainders[remainder] > 1) {
      return true;
    }
  }
  return false;
};

File: JavaWorkSpace/workspace/src/JavaPlayGround/CheckSubarraySum.java
package JavaPlayGround;

import java.util.*;

public class CheckSubarraySum {
    public boolean checkSubarraySum(int[] nums, int k) {
        Map<Integer, Integer> remainders = new HashMap<Integer, Integer>();

        remainders.put(0, 0);

        int total = 0;
        for (int i = 0; i < nums.length; i++) {
            total += nums[i];
            int remainder = total % k;
            if (!remainders.containsKey(remainder)) {
                remainders.put(remainder, i);
            } else if (i - remainders.get(remainder) > 1) {
                return true;
            }
        }
        return false;
    }
}
